components/studentPortal/layout/Sidebar.tsx

```diff
@@ -22,45 +31,52 @@ interface SidebarItemProps {
   page: SidebarPage;
   icon: ReactElement;
   notifications?: boolean;
+  userRole: "freemium" | "student" | "coach";
 }
 const SidebarItem = ({
   name,
   link,
   page,
   icon,
   notifications,
+  userRole,
 }: SidebarItemProps) => {
   const { activePage } = useSelector(activePageSelector);
+  const isDisabled = userRole === "freemium";
+  const href = isDisabled ? "" : link;
+  const onClick = isDisabled ? (e) => e.preventDefault() : undefined;
 
   return (
-    <Link href={link}>
-      <div
-        className={`flex flex-wrap items-center p-4 hover:border-l-4 ${
-          activePage === page
-            ? "border-l-4 border-brandPrimary text-brandPrimary cursor-default"
-            : "cursor-pointer hover:bg-backgroundSecondary"
-        } hover:border-brandPrimary hover:text-brandPrimary `}
-      >
-        <div>
-          {notifications ? (
-            <div className="relative left-6 top-1 ">
-              <div className="flex w-2 h-2 bg-red-500 rounded-full"></div>
-            </div>
-          ) : null}
-          {icon}
+    <>
+      <Link onClick={onClick} href={href}>
+        <div
+          className={`flex flex-wrap items-center p-4 cursor-pointer hover:border-l-4 ${
+            activePage === page ? "border-charmander text-charmander" : ""
+          } hover:border-charmander hover:text-charmander ${
+            isDisabled ? "border-gray-500 text-gray-500" : ""
+          }`}
+        >
+          <div>
+            {notifications ? (
+              <div className="relative left-6 top-1 ">
+                <div className="flex w-2 h-2 bg-red-500 rounded-full"></div>
+              </div>
+            ) : null}
+            {icon}
+          </div>
+          {name}
         </div>
-        {name}
-      </div>
-    </Link>
+      </Link>
+    </>
   );
 };
 
 export const Sidebar: React.FC = () => {
   const { goalApproaching } = useSelector(activePageSelector);
-  const { userRole } = useSelector(profileSelector);
+  const { userRole, createdAt } = useSelector(profileSelector);
   const dispatch = useDispatch();
-
   const { signOut, user } = useAuth();
+  const totalTrialDays = 30;
```

Comment:
for hardcoded constants like this, the convention is to use TOTAL_TRIAL_DAYS instead